### PR TITLE
Move sub-classes of "DataGridViewTopLeftHeaderCell", "FlatButtonAppearance" and "FontCacheEntry" classes to their own files

### PR DESCRIPTION
**Pull request by SergeySmirnov-Akvelon**
## Proposed changes
- Moved sub-classes of "DataGridViewTopLeftHeaderCell", "FlatButtonAppearance" and "FontCacheEntry" classes to their own files 
- Enabled NRT
- Fixed Style issues and code sorting.

## Customer Impact
- No 

## Regression? 
- No

## Risk
- Minimal

## Test environment(s) 
- Microsoft Windows [Version 10.0.19041.388]
- .NET Core SDK 6.0.0-alpha.1.21073.5

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/winforms/pull/4695)

### CODE REVIEW

**Comment by RussKie** [comment]
MC

File: src/System.Windows.Forms/src/System/Windows/Forms/DataGridViewTopLeftHeaderCell.DataGridViewTopLeftHeaderCellRenderer.cs [hunk 1]
```diff
@@ -25,7 +23,7 @@ public static VisualStyleRenderer VisualStyleRenderer
                 {
                     if (visualStyleRenderer is null)
                     {
-                        visualStyleRenderer = new VisualStyleRenderer(HeaderElement);
+                        visualStyleRenderer = new VisualStyleRenderer(s_headerElement);
```
Comment (RussKie) [inline src/System.Windows.Forms/src/System/Windows/Forms/DataGridViewTopLeftHeaderCell.DataGridViewTopLeftHeaderCellRenderer.cs]: Where does this come from?

File: src/System.Windows.Forms/src/System/Windows/Forms/DataGridViewTopLeftHeaderCell.DataGridViewTopLeftHeaderCellRenderer.cs [hunk 2]
```diff
@@ -34,7 +32,7 @@ public static VisualStyleRenderer VisualStyleRenderer
 
             public static void DrawHeader(Graphics g, Rectangle bounds, int headerState)
             {
-                VisualStyleRenderer.SetParameters(HeaderElement.ClassName, HeaderElement.Part, headerState);
+                VisualStyleRenderer.SetParameters(s_headerElement.ClassName, s_headerElement.Part, headerState);
```
Comment (RussKie) [inline src/System.Windows.Forms/src/System/Windows/Forms/DataGridViewTopLeftHeaderCell.DataGridViewTopLeftHeaderCellRenderer.cs:35]: ditto

**Review by RussKie — CHANGES_REQUESTED** [review]
Please clean up the history.

This is also a breaking change, please provide a snip for the breaking docs.

File: src/System.Windows.Forms/src/System/Windows/Forms/DataGridViewTopLeftHeaderCell.DataGridViewTopLeftHeaderCellRenderer.cs (hunk 1 above)
Comment (SergeySmirnov-Akvelon) [reply src/System.Windows.Forms/src/System/Windows/Forms/DataGridViewTopLeftHeaderCell.DataGridViewTopLeftHeaderCellRenderer.cs]: From [this change](https://github.com/dotnet/winforms/pull/4695/files?file-filters%5B%5D=.cs#diff-a9d2d56787df37d056bd0c1ca6b4b9ee2eebfb5f7c645221e9aa12f33187939aR15) to fix the naming

**Comment by SergeySmirnov-Akvelon** [comment]
Fixed history and MC

**Review by RussKie — APPROVED** [review]

File: src/System.Windows.Forms/src/System/Windows/Forms/DataGridViewTopLeftHeaderCell.DataGridViewTopLeftHeaderCellAccessibleObject.cs [hunk 3]
```diff
@@ -158,15 +186,20 @@ public override AccessibleObject Navigate(AccessibleNavigation navigationDirecti
...
             }
 
-            private AccessibleObject NavigateForward()
+            private AccessibleObject? NavigateForward()
             {
-                if (Owner.DataGridView.Columns.GetColumnCount(DataGridViewElementStates.Visible) == 0)
+                if (Owner is null)
+                {
+                    throw new InvalidOperationException(SR.DataGridViewCellAccessibleObject_OwnerNotSet);
+                }
```
Comment (RussKie) [inline src/System.Windows.Forms/src/System/Windows/Forms/DataGridViewTopLeftHeaderCell.DataGridViewTopLeftHeaderCellAccessibleObject.cs]: Just realised - this check here is unnecessary and superficial, this is a private API invoked by `Navigate()` and `FragmentNavigate()` public API that _perform_ the validation.

**Comment by RussKie** [comment]
Docs: https://github.com/dotnet/docs/issues/23529

**Pull request merged**